UnitsNet.Serialization.JsonNet/UnitsNetJsonConverter.cs

```diff
@@ -41,9 +42,36 @@ public class UnitsNetJsonConverter : JsonConverter
 
             object obj = TryDeserializeIComparable(reader, serializer);
             // A null System.Nullable value or a comparable type was deserialized so return this
-            if (!(obj is ValueUnit vu))
+            if (!(obj is ValueUnit) && !(obj is Array))
                 return obj;
 
+            if (obj is Array)
+            {
+                object[] values = (object[]) obj;
+
+                List<object> results = new List<object>();
+                
+                foreach (var value in values)
+                {
+                    object result = ParseValueUnit(value as ValueUnit);
+                    results.Add(result);
+                }
+                
+                Type unitType = objectType.GetElementType();
+                Array typedArray = Array.CreateInstance(unitType, results.Count);
```

Comment:
Is this all actually necessary? I think if you replace the foreach and this below with a LINQ like this, I believe it should work: 
```c#
var results = values.Select(value => ParseValueUnit(value as ValueUnit)).ToArray();
```

Reply:
I think you'll run into runtime exceptions, saying it can't convert object to array or whatever...
This is how I've always converted an "object" into an type specific array (eg string[] or double[] or whatever)

Reply:
You were absolutely right, it wasn't quite as simple as I first imagined. I did make some simplifications in #715 if you want to take a look.